crates/turborepo-scm/src/lib.rs

```diff
@@ -99,36 +99,47 @@ pub(crate) fn wait_for_success<R: Read, T>(
     root_path: impl AsRef<AbsoluteSystemPath>,
     parse_result: Result<T, Error>,
 ) -> Result<T, Error> {
+    if let Err(parse_err) = parse_result {
+        // In this case, we don't care about waiting for the child to exit,
+        // we need to kill it. It's possible that we didn't read all of the output,
+        // or that the child process doesn't know to exit.
+        child.kill()?;
+        let stderr_output = read_git_error_to_string(stderr);
+        let stderr_text = stderr_output
+            .map(|stderr| format!(" stderr: {}", stderr))
+            .unwrap_or_default();
+        let err_text = format!(
+            "'{}' in {}{}{}",
+            command,
+            root_path.as_ref(),
+            stderr_text,
+            parse_err
+        );
+        return Err(Error::Git(err_text, Backtrace::capture()));
+    }
+    // TODO: if we've successfully parsed the output, but the command is hanging for
+    // some reason, we will currently block forever.
```

Comment:
I think we should try to be better about making sure we aren't ever the reason the command is hanging by always reading stdout/stderr until EOF. I think this probably means we read stderr in another thread.

Not super likely that the command ends up hanging because either the stdout or stderr pipe is full, but it's a possibility that we can avoid.

Reply:
In this case, `git status` did hang because `stdout` was full. But, agreed. A reworking of how we handle `stdout`/`stderr` sounds good to me.